### ISSUE TITLE
SpriteSheetAnimation.PingPong = true completes too early

### ISSUE THREAD

**Issue by DavidFidge** [craftworkgames/MonoGame.Extended#804]
If you have 2 frames and set PingPong to true then it should play frame 0, frame 1 then frame 0 again i.e. 3 frames.  It should only become complete when the FrameDuration has expired for the final frame.  Instead it becomes complete as soon as the last frame is hit.  

After investigating I found it is a bug in the AnimationDuration getter.  The correct calculation is (KeyFrames.Length*2 - 1), not (KeyFrames.Length*2 - 2), because you're dealing with actual frames, not frame indexes like the other code further down which has to subtract 2.

Here's a failing unit test:

```
        [Theory]
        [InlineData(0, 0.9f)]
        [InlineData(1, 1f)]
        [InlineData(1, 1.9f)]
        [InlineData(0, 2f)]
        [InlineData(0, 2.9f)]
        public void PingPong_Non_Looping_SpriteSheetAnimation_Should_Return_Correct_Frame_And_Not_Complete(int expectedTextureRegionIndex, float time)
        {
            var textureRegion2D1 = new TextureRegion2D("Region 1", null, new Rectangle());
            var textureRegion2D2 = new TextureRegion2D("Region 2", null, new Rectangle());

            var textureRegions = new[] { textureRegion2D1, textureRegion2D2 };

            var spriteSheetAnimationData = new SpriteSheetAnimationData(
                new[] { 0, 1 },
                1,
                false,
                false,
                true
            );

            var spriteSheetAnimation = new SpriteSheetAnimation("Test", textureRegions, spriteSheetAnimationData);
            var isCompleteFired = false;
            spriteSheetAnimation.OnCompleted += () => isCompleteFired = true;

            spriteSheetAnimation.Play();
            var gameTime = new GameTime(TimeSpan.Zero, TimeSpan.FromSeconds(time));

            spriteSheetAnimation.Update(gameTime);

            Assert.Equal(textureRegions[expectedTextureRegionIndex], spriteSheetAnimation.CurrentFrame);
            Assert.False(spriteSheetAnimation.IsComplete);
            Assert.False(isCompleteFired);
        }
```

**Comment by DavidFidge** [comment]
This one could be subject to a "point of view".

From my point of view the ping pong back to first frame should wait until that frame delay until Complete is fired.  The reason is that when you don't have ping pong set you only get a Complete once the last frame's delay has occurred, thus changing the ping pong Complete functionality would bring it in line with non-ping pong.

HOWEVER for a looping animation you don't want the first frame to be played twice, you want it to keep on ping ponging with no extra delay.

The code would become:

```
        public float AnimationDuration => IsPingPong
            ? (KeyFrames.Length*2 - (IsLooping ? 2 : 1))*FrameDuration
            : KeyFrames.Length*FrameDuration;
```

**Comment by DavidFidge** [comment]
Proposed fix in pull request #806

**Comment by Gandifil** [comment]
Referenced PR has been merged.
